002_simulation_structure/002_communication.tex

```diff
@@ -0,0 +1,205 @@
+\begin{itemize}
+    \item Agents could only talk to their immediate neighbors (+/- 1 floor)
+    \item Agents should be given the ability to ignore communication
+    \item Multiple agents could be “speaking” at a given tick
+    \item One agent could send one message per tick
+    \item \textbf{It is entirely an agent's discretion if/when/how they wish to react to a message as long as their behavior is not deceptive}
+\end{itemize}
+As a team, Team 1 emphasized the last point. The infrastructure should not force any behavior onto the agent as long as the behavior was honest. The basic philosophy of the communication infrastructure was to dictate as little as possible about the agent's behavior while keeping the API friendly, readable and hard to get wrong. This usually meant making providing basic override-able behaviors to be as unobtrusive as possible to team strategies. \newline
+The minimum default message had to have the following fields that are accessible to all the agents that receive it:
+\begin{itemize}
+    \item \textbf{\texttt{MessageType}}: Indicates one of 13 enumerated message types.
+    \item \textbf{\texttt{SenderFloor}}: Returns the floor this message was sent from.
+    \item \textbf{\texttt{TargetFloor}}: The floor the message is addressed to.
+    \item \textbf{\texttt{ID}}: Each message sent is given a unique ID.
+\end{itemize}
+\vspace{\baselineskip}
+For the MVP, Team 1 constructed a system where for example in a scenario where Alice (currently at floor 10) wishes to a message upward:
+\begin{enumerate}
+    \item At Tick 0, Alice would construct the message for Bob containing the following information:
+        \begin{itemize}
+            \item \textbf{Alice's ID}
+            \item \textbf{Alice's current floor}: 30
+            \item \textbf{Message's target floor}: 31 \textit{Since Alice wants to message the floor above her}
+            \item \textbf{Message Type} (to be elaborated on the Common Languages Section)
+        \end{itemize}
+    \item Still at Tick 0, Alice calls \texttt{SendMessage} which takes her message and passes it to the tower.
+    \item The tower acts as the communication authority, finding the agent who is on floor 31 at Tick 0 and inserts the message into the recipient agent's (Bob's) inbox.
+    \item At Tick 1, Bob may choose to call \texttt{ReceiveMessage} which would extract Alice's message (if Alice was the first or only agent to send him a message at Tick 0) and respond.
+    \item If Bob has called \texttt{ReceiveMessage}, Alice's message is removed from Bob's inbox.
+\end{enumerate}
+With this implementation, each agent is responsible for calling \texttt{ReceiveMessage} once per tick to receive any messages. The inbox is FIFO - If multiple messages were received or if there were outstanding messages from previous ticks, only the earliest one can be retrieved at a given tick. \newline
```

Comment:
```suggestion
With this implementation, each agent is responsible for calling \texttt{ReceiveMessage} once per tick to receive any messages. The inbox is FIFO - If multiple messages were received or if there were outstanding messages from previous ticks, only the earliest one can be retrieved at a given tick. 

```
Instead of newline, prefer one additional blank line